appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/TransferAmountPresenter.kt

```diff
@@ -0,0 +1,197 @@
+package org.kinecosystem.appstransfer.presenter
+
+import android.os.Handler
+import org.kinecosystem.appstransfer.view.ITransferAmountView
+import org.kinecosystem.common.base.BasePresenter
+import org.kinecosystem.common.base.Consts
+import org.kinecosystem.common.base.Consts.TRANSACTION_TIMEOUT
+import org.kinecosystem.transfer.model.EcosystemApp
+import org.kinecosystem.transfer.model.getTransactionMemo
+import org.kinecosystem.transfer.model.iconUrl
+import org.kinecosystem.transfer.model.name
+import org.kinecosystem.transfer.repositories.EcosystemAppsRepository
+import org.kinecosystem.transfer.sender.service.SendKinServiceBase
+import org.kinecosystem.transfer.sender.view.TransferBarView
+import org.kinecosystem.transfer.sender.view.TransferInfo
+
+class TransferAmountPresenter(receiverAppName: String, private val senderAppName: String, private val receiverPublicAddress: String, private val repository: EcosystemAppsRepository, private val senderServiceBinder: SenderServiceBinder) : BasePresenter<ITransferAmountView>(), ITransferAmountPresenter, SenderServiceBinder.BinderListener {
```

Comment:
Comments to reduce dependencies / make this reusable.
1. You don't need the EcosystemAppsRepository concrete class - you only need a repository that can get cached balance and store it. Maybe it's even possible to remove this dependency completely and have the service that gets the balance handle storing / caching 
2.